static/app/views/organizationGroupDetails/allEventsTable.tsx

```diff
@@ -99,4 +65,88 @@ const AllEventsTable = (props: Props) => {
   );
 };
 
+type ColumnInfo = {columnTitles: string[]; fields: string[]};
+
+const getColumns = (group: Group, organization: Organization): ColumnInfo => {
+  const isPerfIssue = group.issueCategory === IssueCategory.PERFORMANCE;
+  const isReplayEnabled = organization.features.includes('session-replay-ui');
+
+  const {fields: platformSpecificFields, columnTitles: platformSpecificColumnTitles} =
+    getPlatformColumns(group.project.platform ?? group.platform, {isReplayEnabled});
+
+  const fields: string[] = [
+    'id',
+    'transaction',
+    'title',
+    'release',
+    'environment',
+    'user.display',
+    'device',
+    'os',
+    ...platformSpecificFields,
+    ...(isPerfIssue ? ['transaction.duration'] : []),
+    'timestamp',
+  ];
+
+  const columnTitles: string[] = [
+    t('event id'),
+    t('transaction'),
+    t('title'),
+    t('release'),
+    t('environment'),
+    t('user'),
+    t('device'),
+    t('os'),
+    ...platformSpecificColumnTitles,
+    ...(isPerfIssue ? [t('total duration')] : []),
+    t('timestamp'),
+    t('minidump'),
+  ];
+
+  return {
+    fields,
+    columnTitles,
+  };
+};
+
+const getPlatformColumns = (
+  platform: PlatformKey | undefined,
+  options: {isReplayEnabled: boolean}
+): ColumnInfo => {
+  const replayField = options.isReplayEnabled ? ['replayId'] : [];
+  const replayColumnTitle = options.isReplayEnabled ? [t('replay')] : [];
+
+  const categoryToColumnMap: {[key: string]: ColumnInfo} = {
```

Comment:
A stricter way to type this would be like this:

```suggestion
  const categoryToColumnMap: Record<PlatformCategory, ColumnInfo> = {
```

One caveat is that you'll have to add an entry for `[PlatformCategory.SERVERLESS]` or the linter will complain, but that's proof that TS is properly doing its job here 😄

Reply:
I like this! forgot about using record, thanks!

Reply:
Hmm, this actually doesn't work straight up like this, lemme dig into it some more
<img width="813" alt="image" src="https://user-images.githubusercontent.com/44422760/203122291-d1570383-7f19-4bd4-bfe5-65e74154172f.png">

Reply:
Yep it's because you have to add `[PlatformCategory.SERVERLESS]` as an entry, which is `3` in the enum.

Reply:
Would be nice if vscode showed the enum instead of the value, got confused for a sec lol, but it makes sense.

Reply:
Agreed, I only know this because I've been confused by this in the past haha

One way to make it cleaner when linting is use string values in the enum, like 
```
enum PlatformCategory = {
  MOBILE = 'mobile',
  FRONTEND = 'frontend',
  ...
}
```